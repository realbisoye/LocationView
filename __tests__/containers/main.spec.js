import React from 'react'
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import MainContainer from '../../app/container/main'

const mockStore = configureStore([thunk])

const initialState = {
  markers: [],
}
let store

describe('Main Container', () => {
  beforeEach(() => {
    store = mockStore(initialState)
  })

  it('render main container', () => {
    const rendered = renderer.create(
      <Provider store={store}>
        <MainContainer />
      </Provider>)

    expect(rendered).toMatchSnapshot()
  })
})
