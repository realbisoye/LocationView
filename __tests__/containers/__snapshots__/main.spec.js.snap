// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Container render main container 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "right": 112.5,
        },
      ]
    }
  >
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "flex": 1,
            },
            undefined,
            undefined,
            undefined,
          ]
        }
      >
        <RCTSafeAreaView
          style={
            Object {
              "backgroundColor": "#C2E098",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "maxHeight": "5%",
                  "paddingHorizontal": 20,
                  "top": 0,
                  "width": "100%",
                },
                Object {
                  "backgroundColor": "#C2E098",
                },
              ]
            }
          >
            <View
              accessible={true}
              isTVSelectable={true}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "opacity": 1,
                }
              }
              testID="menuButton"
            >
              <Text
                allowFontScaling={false}
                style={
                  Array [
                    Object {
                      "color": "#171914",
                      "fontSize": 25,
                    },
                    undefined,
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                
              </Text>
            </View>
            <Text
              style={
                Object {
                  "fontSize": 22,
                  "fontWeight": "bold",
                  "marginLeft": "22%",
                  "zIndex": 1000,
                }
              }
              testID="apptitle"
            >
              Markers
            </Text>
          </View>
          <RCTScrollView
            contentContainerStyle={
              Object {
                "borderColor": "#88BF41",
                "borderTopWidth": 2,
                "flex": 1,
                "marginTop": 20,
              }
            }
          >
            <View />
          </RCTScrollView>
          <Modal
            animationType="slide"
            hardwareAccelerated={false}
            onRequestClose={[Function]}
            transparent={false}
            visible={false}
          >
            <RCTSafeAreaView
              style={
                Object {
                  "backgroundColor": "#C2E098",
                  "flex": 1,
                  "flexDirection": "column",
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "flex": 1,
                      "flexDirection": "row",
                      "maxHeight": "5%",
                      "paddingHorizontal": 20,
                      "top": 0,
                      "width": "100%",
                    },
                    Object {
                      "backgroundColor": "#C2E098",
                    },
                  ]
                }
              >
                <View
                  accessible={true}
                  isTVSelectable={true}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "opacity": 1,
                    }
                  }
                  testID="menuButton"
                >
                  <Text
                    allowFontScaling={false}
                    style={
                      Array [
                        Object {
                          "color": "#171914",
                          "fontSize": 25,
                        },
                        undefined,
                        Object {
                          "fontFamily": "Material Design Icons",
                          "fontStyle": "normal",
                          "fontWeight": "normal",
                        },
                        Object {},
                      ]
                    }
                  >
                    
                  </Text>
                </View>
                <Text
                  style={
                    Object {
                      "fontSize": 22,
                      "fontWeight": "bold",
                      "marginLeft": "22%",
                      "zIndex": 1000,
                    }
                  }
                  testID="apptitle"
                >
                  Add Marker
                </Text>
              </View>
              <View
                style={
                  Object {
                    "flex": 1,
                    "marginHorizontal": 20,
                    "marginTop": 30,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "column",
                      "marginVertical": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#171914",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "marginBottom": 5,
                      }
                    }
                  >
                    Title
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    autoCapitalize="words"
                    onChangeText={[Function]}
                    style={
                      Object {
                        "borderColor": "#88BF41",
                        "borderWidth": 1,
                        "height": 40,
                        "paddingHorizontal": 10,
                      }
                    }
                    underlineColorAndroid="transparent"
                  />
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "column",
                      "marginVertical": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#171914",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "marginBottom": 5,
                      }
                    }
                  >
                    Description
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    onChangeText={[Function]}
                    style={
                      Object {
                        "borderColor": "#88BF41",
                        "borderWidth": 1,
                        "height": 40,
                        "paddingHorizontal": 10,
                      }
                    }
                    underlineColorAndroid="transparent"
                  />
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "column",
                      "marginVertical": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#171914",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "marginBottom": 5,
                      }
                    }
                  >
                    Latitude
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    keyboardType="numeric"
                    onChangeText={[Function]}
                    style={
                      Object {
                        "borderColor": "#88BF41",
                        "borderWidth": 1,
                        "height": 40,
                        "paddingHorizontal": 10,
                      }
                    }
                    underlineColorAndroid="transparent"
                  />
                </View>
                <View
                  style={
                    Object {
                      "flexDirection": "column",
                      "marginVertical": 10,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#171914",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "marginBottom": 5,
                      }
                    }
                  >
                    Longitude
                  </Text>
                  <TextInput
                    allowFontScaling={true}
                    keyboardType="numeric"
                    onChangeText={[Function]}
                    style={
                      Object {
                        "borderColor": "#88BF41",
                        "borderWidth": 1,
                        "height": 40,
                        "paddingHorizontal": 10,
                      }
                    }
                    underlineColorAndroid="transparent"
                  />
                </View>
                <View>
                  <View
                    accessibilityRole="button"
                    accessibilityStates={
                      Array [
                        "disabled",
                      ]
                    }
                    accessible={true}
                    isTVSelectable={true}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Array [
                          Object {},
                          Object {},
                        ]
                      }
                    >
                      <Text
                        disabled={true}
                        style={
                          Array [
                            Object {
                              "color": "#007AFF",
                              "fontSize": 18,
                              "padding": 8,
                              "textAlign": "center",
                            },
                            Object {
                              "color": "#BF41BA",
                            },
                            Object {
                              "color": "#cdcdcd",
                            },
                          ]
                        }
                      >
                        Add
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
            </RCTSafeAreaView>
          </Modal>
        </RCTSafeAreaView>
      </View>
      <RCTSafeAreaView
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "flex": 1,
            }
          }
        />
      </RCTSafeAreaView>
    </View>
  </View>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
        "height": 1334,
        "left": 0,
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "transform": Array [
          Object {
            "translateX": 0,
          },
        ],
        "width": 750,
      }
    }
  >
    <RCTSafeAreaView
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#88BF41",
          "flex": 1,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "row",
              "maxHeight": "5%",
              "paddingHorizontal": 20,
              "top": 0,
              "width": "100%",
            },
            Object {
              "backgroundColor": "#88BF41",
            },
          ]
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID="menuButton"
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#171914",
                  "fontSize": 25,
                },
                undefined,
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            
          </Text>
        </View>
        <Text
          style={
            Object {
              "fontSize": 22,
              "fontWeight": "bold",
              "marginLeft": "22%",
              "zIndex": 1000,
            }
          }
          testID="apptitle"
        >
          Location View
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "width": "100%",
          }
        }
      >
        MapView
      </View>
    </RCTSafeAreaView>
  </View>
</View>
`;
